backend/charm/src/charm.py

```diff
@@ -113,6 +113,7 @@ def _migrate_database(self):
             logger.error(e.stdout)
             logger.error(e.stderr)
             self.unit.status = BlockedStatus("Database migration failed")
+            raise SystemExit(0)
```

Comment:
I would like to suggest using `sys.exit(0)` as it’s used in the docs: https://docs.python.org/3/library/exceptions.html#SystemExit

Reply:
I think both can be used interchangeably. The docs doesn't seem to recommend one over the other. Personally, what I like about  `raise SystemExit(0)` is that it's more clear that it breaks flow and not just a regular function call. But don't really mind either.